### PR TITLE
Update to istio 1.11

### LINKED ISSUE #1979: Manifests not ready for K8s 1.22
/kind bug

With the new Kubernetes 1.22 release, some APIs have been removed: https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/

Many manifests in this repository are still using some of these APIs and won't work for this release. 

For example, the following CRDs are still using `apiextensions.k8s.io/v1beta1` and are causing issues with the Charmed Kubeflow distribution:
```
authcodes.dex.coreos.com
sparkapplications.sparkoperator.k8s.io
```

### PR DESCRIPTION
**Pull request by chingandy**
**Which issue is resolved by this Pull Request:**
Resolves #1979

**Description of your changes:**
This will update istio to v1.11.0 in order to get Kubeflow able to install on Kubernetes 1.22.
Changes are:
1. Generate istio-1-11 manifests and add them to their package
2. Update kustomization.yaml in example
3. Fix the envoyfilter resource yaml in common/oidc-authservice/base/envoy-filter.yaml, which is required to make kubeflow actually work with Istio 1.11

**Checklist:**
- [ ] Unit tests pass:
  **Make sure you have installed kustomize == 3.2.1**
    1. `make generate-changed-only`
    2. `make test`

### CODE REVIEW

File: common/istio-1-11/cluster-local-gateway/base/gateway-authorizationpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  name: cluster-local-gateway
+spec:
+  action: ALLOW
+  selector:
+    # Same as the cluster-local-gateway Service selector
+    matchLabels:
+      app: cluster-local-gateway
+      istio: cluster-local-gateway
+  rules:
+  - {}
```
Comment (mvoitko) [inline common/istio-1-11/cluster-local-gateway/base/gateway-authorizationpolicy.yaml:14]: NO newline at the EOF? why have a list item with empty map?

File: common/istio-1-11/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# Istio
+
+## Upgrade Istio Manifests
+
+Istio ships with an installer called `istioctl`, which is a deployment /
```
Comment (mvoitko) [inline common/istio-1-11/README.md:5]: Could the installation be done only with kustomizations?

Comment (chingandy) [reply common/istio-1-11/README.md:5]: yes, with the single command stated [here](https://github.com/kubeflow/manifests#install-with-a-single-command)

**Comment by kimwnasptd** [comment]
thank you for the PR @chingandy! I also tried it out and everything works fine and the files are all in the correct places.

I also checked the upgrade instructions from going to 1.9 to 1.10 and then to 1.11 and we are good to go!
https://istio.io/latest/news/releases/1.10.x/announcing-1.10/upgrade-notes/
https://istio.io/latest/news/releases/1.11.x/announcing-1.11/upgrade-notes/

/lgtm
/approve

**Pull request merged**